samples/GetUserProfile.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+# Get a users display name and avatar
+# Args: host:port username password user_id
+# Error Codes:
+# 2 - Could not find the server.
+# 3 - Bad URL Format.
+# 4 - Bad username/password.
+# 11 - Wrong room format.
+# 12 - Couldn't find room.
+
+
+import sys
+sys.path.insert(0, "../") # add ../ to PYTHONPATH
+
+from matrix_client.client import MatrixClient, User
+from matrix_client.api import MatrixRequestError
+from requests.exceptions import MissingSchema
+from getpass import getpass
+
+host = ""
+username = ""
+room = ""
+
+if len(sys.argv) > 1:
+    host = sys.argv[1]
+else:
+    host = input("Host (ex: http://localhost:8008 ): ")
+
+client = MatrixClient(host)
+
+if len(sys.argv) > 2:
+    username = sys.argv[2]
+else:
+    username = input("Username: ")
+
+if len(sys.argv) > 3:
+    password = sys.argv[3]
+else:
+    password = getpass() #Hide user input
+
+try:
+    client.login_with_password(username,password)
+except MatrixRequestError as e:
+    print(e)
+    if e.code == 403:
+        print("Bad username or password.")
+        sys.exit(4)
+    else:
+        print("Check your sever details are correct.")
+        sys.exit(3)
+
+except MissingSchema as e:
+    print("Bad URL format.")
+    print(e)
+    sys.exit(2)
+
+
+userid = None
```

Comment:
As above, you don't need to declare up front since all branches below set it (and branches share the same scope)